### PR TITLE
completed task 5 get-api-articles

### PR DESCRIPTION
**Pull request by kylexpowis**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -35,3 +35,26 @@ describe("/api", () => {
...
     });
 });
+describe("/api/articles", () => {
+    test("GET: 200 sends an array of article objects", () => {
+      return request(app)
+        .get("/api/articles")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeInstanceOf(Array);
```
Comment (Keb84) [inline __tests__/app.test.js:48]: When asserting inside a forEach you should be doing a length check first 👀

File: __tests__/app.test.js [hunk 2]
```diff
@@ -35,3 +35,26 @@ describe("/api", () => {
...
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(String)
+            })
+        })
+    })
```
Comment (Keb84) [inline __tests__/app.test.js:61]: You should be checking the articles are coming back in created_at order descending by default. Please add a test for this. You can use the `jest -sorted` library for this 👀

**Review by Keb84 — CHANGES_REQUESTED** [review]
Nice work. Please see my comments, make the changes and resubmit 🙂

File: __tests__/app.test.js [hunk 3]
```diff
@@ -35,3 +36,37 @@ describe("/api", () => {
...
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: expect.any(String),
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              article_img_url: expect.any(String),
+              comment_count: expect.any(String)
+            })
+        })
```
Comment (nc-hev-mag) [inline __tests__/app.test.js:60]: Would be good to check that there is no key of body here too as toMatchObject is a flexible matcher and wouldn't pick up on that if it was included

File: controllers/articles.controllers.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    });
+};
+
+exports.getArticles = (req, res, next) => {
+    selectArticles()
+    .then((articles) => {
+        res.status(200).send({ articles });
+    })
+    .catch((err) => {
+        console.log(err);
```
Comment (nc-hev-mag) [inline controllers/articles.controllers.js:21]: remove console log

File: models/articles.models.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+exports.selectArticles = () => {
+    let queryString = `SELECT articles.author,
+    articles.title,
+    articles.article_id,
+    articles.topic,
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url,
+    COUNT(comments.comment_id) AS comment_count
```
Comment (nc-hev-mag) [inline models/articles.models.js:22]: nice SQL

**Review by nc-hev-mag — APPROVED** [review]
Hey Kyle! Couple of pointers, but good to merge once you've taken a look and removed that console.log 🎉 Make sure your endpoints.json is up to date for this endpoint too 👍

**Pull request closed without merging**